src/components/__tests__/ImageCarousel.spec.tsx

```diff
@@ -0,0 +1,55 @@
+import { render, RenderAPI } from '@testing-library/react-native'
+import React from 'react'
+import { ReactTestInstance } from 'react-test-renderer'
+
+import wrapWithTheme from '../../testing/wrapWithTheme'
+import ImageCarousel from '../ImageCarousel'
+
+jest.mock('react-native/Libraries/Image/Image', () => {
+  return {
+    ...jest.requireActual('react-native/Libraries/Image/Image'),
+    getSize: (uri: string, success: (w: number, h: number) => void) => {
+      success(1234, 1234)
+    }
+  }
+})
+
+describe('ImageCarousel ', () => {
+  const images = [
+    {
+      id: 0,
+      image: 'Arbeitshose'
+    },
+    {
+      id: 1,
+      image: 'Arbeitsschuhe'
+    }
+  ]
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const renderImageCarousel = (): RenderAPI => {
+    return render(<ImageCarousel images={images} />, { wrapper: wrapWithTheme })
+  }
+
+  const getUri = (image: ReactTestInstance): string => image.props.source[0].uri
+
+  it('should render images', async () => {
+    const { getByTestId, findAllByRole } = renderImageCarousel()
+
+    const row = getByTestId('Swipeable')
+    expect(row).toBeTruthy()
+    const swipeable = row.children[0] as ReactTestInstance
+    swipeable.instance.loadImage(1) // preload the second image
```

Comment:
The image rendering library we use renders all images next to each other. Images that haven't been rendered yet are only represented by an empty `View`, meaning they are only displayed after loadImage is called for that image. Therefore after the second image was loaded, there are two images, As the first one doesn't get 'unloaded' or sth. 
Therefore testing swiping is rather difficult here, so I just tested wether there are two images.

Reply:
Perhaps you could add a check that before `loadImage` only one image is rendered? I think a short comment in the code to explain this would also be nice.